### ISSUE TITLE
Backspace erases all entered text

### ISSUE THREAD

**Issue by n1kk** [akinomyoga/ble.sh#104]
**0.4.0-devel3+032f6b2**
Windows 10 **4.4.23(1)-release, x86_64-pc-msys**

Backspace erases all entered text on windows host. 

I installed it on my linux machine with Fedora 34 and it works fine there.

Is it a bug or is it configuration?

**Comment by akinomyoga** [comment]
Thank you for the report! I guess your terminal sends `C-_` for the <kbd>Backspace</kbd>. `C-_` is bounded to "undo" so the previous consecutive input of text is canceled. I think we should add detection of such a terminal.

- **Q1**. What is your terminal in Windows 10?
- **Q2**. Could you send me the result of the following commands in Windows 10?
```bash
$ declare -p TERM _ble_term_DA1R _ble_term_DA2R
```
- **Q3**. What will be input by pressing <kbd>C-v</kbd><kbd>Backspace</kbd> or <kbd>C-q</kbd><kbd>Backspace</kbd> in your terminal in Windows 10?

**Comment by n1kk** [comment]
-  If you're asking about the application then standalone I use Fluent Terminal, but I also use terminals in Webstorm and VS Code Ides, and it behaves same in all of them.

-  ```bash
    ❯   declare -p TERM _ble_term_DA1R _ble_term_DA2R
    declare -x TERM="xterm-256color"
    declare -- _ble_term_DA1R=""
    declare -- _ble_term_DA2R=""
    ```
- How exactly I trigger those shortcuts exactly? Is it `shift` + `c` then `v` then `Backspace` ? If I do it fast it erases, if I do it slower then it turns into this `❯ Cv [_ble_term_cvvis] `

**Comment by akinomyoga** [comment]
Thank you for the information. In my Windows, there is no problem with VSCode terminal, but I have never tried Fluent Terminal and Webstorm. I'll take a look at them.

> How exactly I trigger those shortcuts exactly? 

<kbd>C-v</kbd><kbd>Backspace</kbd> means to press Control+v and then press Backspace.

**Comment by akinomyoga** [comment]
I've installed Fluent Terminal, but I couldn't reproduce the problem. I have a question.

- **Q4**. To which system do you connect to from these terminals? Is it WSL2?

Since you consistently experience the same behavior in all the terminals, maybe it is related to some user settings. I have an additional question.

- **Q5**. Does the problem reproduce in the plain Bash started by the following command?
```bash
$ bash --noprofile --norc            # <-- Here, start another Bash session inside Bash
$ source /path/to/blesh/out/ble.sh   # <-- Please replace "/path/to/blesh/out/ble.sh" with your "ble.sh" position
$                                    # <-- Then input something and press Backspace
```

Also, I still would like to see the result of Q3.

**Comment by akinomyoga** [comment]
OK! I could reproduce it with MSYS2! It seems to be related to the default `.inputrc` of MSYS2. I'll check what's happening here.

**Comment by n1kk** [comment]
Yes, it's system wide so I don't think it's a terminal problem, and if I disable ble.sh it's not reproduceable, that's why I thought it's somehow a ble.sh behavior.

`C-v` `Backspace` after typing `ls`
```bash
❯ ls^?
```

I use plain bash that comes with git instalation.

**Comment by akinomyoga** [comment]
It seems like the bug of MSYS2 .inputrc. I'm going to ask this on the msys2 mailing list.

For the time being you can comment out the following line in `~/.inputrc` (L79):

```inputrc
  "\C-?": backward-kill-line         # Ctrl-BackSpace
```

Or, you may entirely disable the `.inputrc` by putting the following line at the top of `~/.bashrc`:

```bash
INPUTRC=/dev/null
```

**Comment by akinomyoga** [comment]
I've posted an email in the list `msys2-users`. Let's wait to see what comes out:

- [MSYS2 / Thread: [Msys2-users] MSYS2 .inputrc contains an unintended and ineffective keybinding which causes a problem](https://sourceforge.net/p/msys2/mailman/msys2-users/thread/CAFLRLk-UX7S%3DTAerNix7HvxDAv4aY2FwZuFZz%3DU%2BTLBAWxCLEg%40mail.gmail.com/#msg37275646)

**Comment by akinomyoga** [comment]
I added a workaround for this issue in commit 9e786ae, in which the keybinding `"\C-?": backward-kill-line` is canceled when it was detected in MSYS systems. Now <kbd>Backspace</kbd> should work as expected without modifying `.inputrc` nor `.bashrc`. Thank you!

**Comment by akinomyoga** [comment]
I sent a PR to MSYS2 (https://github.com/msys2/MSYS2-packages/pull/2490), and it was merged

> I use plain bash that comes with git instalation.

Thanks for the information. It seems Git for Windows is based on MSYS2 but also has its own configurations. I have sent another PR to Git for Windows (https://github.com/git-for-windows/build-extra/pull/341).

**Comment by akinomyoga** [comment]
The PR to Git for Windows was also merged just now. Now I think everything is fine.

@n1kk Is there any other remaining problems related to this? If so, I think we can now close the issue. Thank you.

**Comment by akinomyoga** [comment]
@n1kk Thank you!
